shared/chat/conversation/list/index.desktop.js

```diff
@@ -1,244 +1,269 @@
+class BaseList extends Component<void, Props, State> {
+  _cellCache = new Virtualized.CellMeasurerCache({
+    fixedWidth: true,
+    keyMapper: (rowIndex: number) => this.props.messageKeys.get(rowIndex),
+  })
 
-    this._setupDebug()
-  }
+  _list: any
+  _keepIdxVisible: number = -1
+  _lastRowIdx: number = -1
 
-  _setupDebug () {
-    if (__DEV__ && typeof window !== 'undefined') {
-      window.dumpChat = (...columns) => {
-        console.table(this.state.messages.toJS().map(m => ({
-          ...m,
-          decoded: m.message && m.message.stringValue(),
-        })), columns.length && columns)
-      }
-    }
-  }
+  // This is similar to a state but it changing value doesn't actually need to cause us to re-render.
+  // We're either locked at the bottom or not and it only affects when we get new messages which'll cause a render anyways
+  _scrollTop = 0
 
-  shouldComponentUpdate (nextProps: Props, nextState: State) {
-    return !shallowEqual(this.props, nextProps) || !shallowEqual(this.state, nextState)
+  state = {
+    isLockedToBottom: true,
+    listRerender: 0,
+    selectedMessageKey: null,
   }
 
-  componentWillUnmount () {
-    // Stop any throttled/debounced functions
-    this._onScroll.cancel()
-    this._recomputeListDebounced.cancel()
-    this._onScrollSettled.cancel()
-  }
+  _onAction = (message: Constants.ServerMessage, event: any) => { throw new Error('_onAction Implemented in PopupEnabledList') }
+  _onShowEditor = (message: Constants.Message, event: any) => { throw new Error('_onShowEditor Implemented in PopupEnabledList') }
+  _onEditLastMessage = () => { throw new Error('_onEditLastMessage Implemented in PopupEnabledList') }
 
-  componentWillUpdate (nextProps: Props, nextState: State) {
-    // If a message has moved from pending to sent, tell the List to discard
-    // heights for it (which will re-render it and everything after it)
-    // TODO this doesn't work for things that take a bit to load (imgs)
-    if (this._toRemeasure.length) {
-      this._toRemeasure.forEach(item => {
-        this._cellCache.clearRowHeight(item)
-        if (this._listIsGood()) {
-          this._list.recomputeRowHeights(item)
-        }
-      })
-      this._toRemeasure = []
+  componentDidUpdate (prevProps: Props, prevState: State) {
+    // Force a rerender if we passed a row to scroll to. If it's kept around the virutal list gets confused so we only want it to render once basically
+    if (this._keepIdxVisible !== -1) {
+      this.setState({listRerender: this.state.listRerender + 1}) // eslint-disable-line react/no-did-update-set-state
+      this._keepIdxVisible = -1
     }
+    this._lastRowIdx = -1 // always reset this to be safe
 
-    if (this._shouldForceUpdateGrid) {
-      this._shouldForceUpdateGrid = false
-      if (this._listIsGood()) {
-        this._list.forceUpdateGrid()
-      }
+    if (this.props.editLastMessageCounter !== prevProps.editLastMessageCounter) {
+      this._onEditLastMessage()
```

Comment:
this is cause we're doing dom operations and we want to do it after the render. same as the old impl